ios/swift/v2/AmpliSwiftSampleApp/Shared/Plugins/LoggingPlugin.swift

```diff
@@ -0,0 +1,25 @@
+import Foundation
+import AmplitudeSwift
+
+class LoggingPlugin: Plugin {
+    let type: PluginType
+    var amplitude: Amplitude?
+    let jsonEncoder: JSONEncoder
+
+    init() {
+        self.type = .before
+        self.jsonEncoder = JSONEncoder()
+        self.jsonEncoder.outputFormatting = [.prettyPrinted]
+    }
+
+    func setup(amplitude: Amplitude) {
+        self.amplitude = amplitude
```

Comment:
Do we need this reference?

Reply:
Base `Plugin` requires `amplitude` property. Maybe it is unnecessary (`var` with public setter, for example) and should be re-factored.
https://github.com/amplitude/Amplitude-Swift/blob/main/Sources/Amplitude/Types.swift#L107
```
public protocol Plugin: AnyObject {
    var type: PluginType { get }
    var amplitude: Amplitude? { get set }
    func setup(amplitude: Amplitude)
    func execute(event: BaseEvent?) -> BaseEvent?
}
```

Reply:
Thanks. Since not all Plugins use the amplitude instance, I think it would be good to remove this requirement while the SDK is still in beta. We made similar improvements to simplify Plugin creation in Browser 2.0.

@liuyang1520 @kevinpagtakhan wdyt?

```diff
public protocol Plugin: AnyObject {
    var type: PluginType { get }
-    var amplitude: Amplitude? { get set }
    func setup(amplitude: Amplitude)
    func execute(event: BaseEvent?) -> BaseEvent?
}
```

Reply:
Removing this var requirement sounds good to me.

In this line specifically, I think we don't need to implement `setup` as we have the [default implementation](https://github.com/amplitude/Amplitude-Swift/blob/8f0e80f789bd4e6e383876330eb7e418ca2cdccb/Sources/Amplitude/Types.swift#L126-L128).

Reply:
@justin-fiedler @liuyang1520 
https://github.com/amplitude/Amplitude-Swift/pull/73